department-of-veterans-affairs/caseflow
# Kev ma/appeals 29212

Resolves [APPEALS-29212](https://jira.devops.va.gov/browse/APPEALS-29212)

# Description
When the user selects the Reassign Package in the "Request package action dropdown", a modal will popup prompting the user for a reason for the request. The request cannot be made until a reason has been provided. When a request is submitted, a ReassignPackageTask is created in the backend that inherits from ReviewPackageTask, and is assigned the ReviewPackageTask as the parent task. The ReviewPackageTask is assigned the new ReassignPackageTask as the child task, and its status is changed to "on_hold"

## Acceptance Criteria

## Testing Plan
1. Signin as JOLLY_POSTMAN
2. Go to "/queue/correspondence" and select any veteran
3. Click on the "Request package action" dropdown and click on "Reassign Package"
4. The Reassign Package Modal should be displayed with "Confirm Request" button disabled
5. Type sample text into the text field; note that the "Confirm Request" button is now enabled
6. Open the console by inspecting the page or hitting F12 and go to the Network tab. 
7. Double-check that the Network tab is recording network activity
8. On the modal, hit confirm request. The Network tab should record activity. Click on the row that says "task" and then hit the response tab to see a status 200 as seen in the image below
<img width="763" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/07c90047-1b8e-4447-8f9a-50b30b59de71">
9. Open up the rails console, and search for the ReassignPackageTask
<img width="996" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/5790a301-22d5-47d9-b3da-c9d4d5b3e11f">
10. Verify that it has access to the parent task, the parent task has access to the child task
<img width="997" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/20928330-7172-4496-93c0-c6230563a62d">

11. Verify that parent task is ReviewPackageTask with a status of 'on_hold'
<img width="997" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/f94b4f12-8c31-41a3-a418-50eccded81c0">




# Frontend
## User Facing Changes

 BEFORE|AFTER
 <img width="608" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/3aec1702-63ec-4d3f-8a9d-7ebc6e3887e4">|
<img width="608" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/104021955/7db82f45-7d82-4bbf-8e83-0b5d825e7ef1">


## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
